src/compile_time_environment.cpp

```diff
@@ -13,14 +13,29 @@ CompileTimeEnvironment::CompileTimeEnvironment() {
 	declare_builtin("array_extend");
 	declare_builtin("array_join");
 
-	declare_builtin("+");
-	declare_builtin("-");
-	declare_builtin("*");
-	declare_builtin("/");
-	declare_builtin("<");
-	declare_builtin("=");
-	declare_builtin("==");
-	declare_builtin(".");
+	// TODO: refactor, figure out a nice way to build types
+	auto var_mono_id = m_typechecker.new_var();
+	auto var_id = m_typechecker.m_core.mono_data[var_mono_id].data_id;
+
+	// TODO: i use the same mono thrice... does this make sense?
+	auto term_mono_id = m_typechecker.m_core.new_term(
+		TypeChecker::BuiltinType::Function,
+		{var_mono_id, var_mono_id, var_mono_id},
+		"[builtin] (a, a) -> a");
```

Comment:
I would think so... This represents a binary operation on a set that is represented by the type, and the type is not known so we use a var... Shouldn't it be a poly? And later on do an Inst for each use? Else the deduction will fail if it's used for different types... Right?

Reply:
Ah but it's a term that takes 3 vars as arguments, I think this is correct. We'll see

Reply:
a poly is created from this mono on the very next line

Reply:
Ah, now I get it